lib/chef/resource/yum_package.rb

```diff
@@ -30,13 +30,13 @@ class YumPackage < Chef::Resource::Package
       property :flush_cache, Hash, default: { before: false, after: false }, coerce: proc { |v|
         # TODO these append rather than set. This is probably wrong behavior, but we're preserving it until we know
         if v.is_a?(Array)
-          v.each { |arg| flush_cache[arg] = true }
-          flush_cache
+          x = {}
+          v.each { |arg| x[arg] = true }
+          x
         elsif v.any?
           v
         else
-          # TODO calling flush_cache({}) does a get instead of a set. This is probably wrong behavior, but we're preserving it until we know
-          flush_cache
+          { before: v, after: v}
```

Comment:
shouldn't this just be v?  although i think it makes my brain hurt...   i don't think anything you do here turns this from a getter into a setter does it?

i suspect nobody uses flush_cache({}) as a no-op getter.

this code is now generally trying way to hard to be backcompat and screwing up everything in the process, i'd sort of scrap it and rewrite it cleanly, then ship fixes if we actually do break anyone (because currently we've broken /everyone/ and the code is very hard to follow).